### DISCUSSION TITLE
Level2 processing - 100% cloud-cover use case

### DISCUSSION THREAD

**Discussion by davedavemckay** [davidfrantz/force#284] [Ideas]
Hi, I'm working on a project that makes a positive use of cloud-cover QAI data. One person's trash is another person's treasure and all that - they're not interested in the satellite image, they just want the clouds.
My thinking is I can process time series of Sentinel-2 L1C data with FORCE and use the *.qai files to make a dataset for them (I also need the level2 data for other projects, so would do full processing).
I wondered - is it possible to produce level2 data even when the L1C data shows 100 % cloud cover? There is a temptation to set missing data values to 100 % cloud, but I'd prefer to have confirmed 100 % cloud and leave missing data values to true gaps in the data.
Thanks,
Dave

**Comment by ernstste** [comment]
Hi Dave,

I'm not sure i'm following 100%, but my suggestion for giving this a try would be using the following settings:
```
ERASE_CLOUDS = FALSE
MAX_CLOUD_COVER_FRAME = 100
MAX_CLOUD_COVER_TILE  = 100
CLOUD_BUFFER  = 0
CIRRUS_BUFFER = 0
```

This will ensure you're not setting pixels to nodata where FORCE has detected clouds and will make sure we don't create a buffer around the clouds. Depending on your application, you may want to consider using a buffer. 
I can't recall if the max allowed cloud threshold is inclusive or exclusive though, @davidfrantz would have to confirm here.

Some general information: the cloud detection algorithm favors an overestimation of clouds over an underestimation as it is the lesser evil in time series analysis. Still, you are likely going to see edges of clouds etc. which haven't been caught when setting the buffer to 0.


>  is it possible to produce level2 data even when the L1C data shows 100 % cloud cover?

I'm not sure if you refer to the cloud estimations of Level-1 data provided by the Landsat/Sentinel-2 metadata. If so - this is completely irrelevant to the FORCE processing. It is only used for selecting data to download when using force-level1-landsat or force-level1-csd.

Stefan

**Comment by davidfrantz** [comment]
Hi Dave,

Stefan is right. 

```
MAX_CLOUD_COVER_FRAME = 100
MAX_CLOUD_COVER_TILE  = 100
```

This will also produce L2 data for 100% cloud images. But please note that the reflectance type is "surface reflectance", i.e. it considers the full path through the atmosphere to the ground (2x). Thus, I guess that the reflectance over a cloud is physically not correct (if that is of concern to you).

The usage of the buffers depends on how you want to proceed exactly (i.e. whether you want to analyze pixels or log files).
If you want to use the data for both purposes, I would recommend you to use the buffer. There is a separate bit flag for the buffer, thus it 
is possible to analyze core-cloud and buffered-cloud areas separately. However, please keep in mind that the FORCE-generated log file will report cloud coverage as the sum of core-cloud, cloud-buffer, cloud-shadow.

Cheers,
David

**Comment by davedavemckay** [comment]
Excellent, thank you both for your replies.
Very useful indeed.
Cheers,
Dave
